src/components/layout/nav/nav-desktop.tsx

```diff
@@ -0,0 +1,239 @@
+
+const NowrapButton = styled(Button)`
+    white-space: nowrap;
+`
+const NavGetTrading = styled.div`
+    display: inline-flex;
+    text-align: right;
+    align-items: center;
+    justify-content: center;
+    padding: 0;
+`
+const StyledButton = styled.span<{ active: boolean }>`
+    ${SharedLinkStyle};
+    cursor: pointer;
+    user-select: none;
+    white-space: nowrap;
+    @media ${device.laptopM} {
+        padding: 0.5rem 0.8rem;
+    }
+`
+const SignupButton = styled(Button)`
+    margin-left: 1.6rem;
+    opacity: 0;
+`
+const NavCenter = styled.ul`
+    text-align: center;
+    padding: 0;
+    display: flex;
+    justify-content: center;
+    line-height: 1.2;
+    @media (max-width: 1227px) {
+        font-size: 14px;
+    }
+    @media (max-width: 1175px) {
+        font-size: 12px;
+    }
+    @media (max-width: 1116px) {
+        font-size: 11px;
+    }
+    @media (max-width: 991px) {
+        display: none;
+    }
+`
+const NavLeftMain = styled.div`
+    text-align: left;
+    display: flex;
+    align-items: center;
+    max-width: 30rem;
+    width: 100%;
+`
+
+const query = graphql`
+    query {
+        deriv: file(relativePath: { eq: "logo.png" }) {
+            ...fadeIn
+        }
+    }
+`
+
+const LanguageSwitcherNavDesktop = ({ no_language }: { no_language: boolean }) =>
+    !no_language && <LanguageSwitcher short_name="true" is_high_nav />
+
+const NavDesktop = ({
+    base,
+    is_ppc,
+    is_ppc_redirect,
+    is_logged_in,
+    hide_signup_login,
+    academy_logo,
+    no_language,
+}: NavTypes) => {
+    const data = useStaticQuery(query)
+    const button_ref = useRef(null)
+    const navigation_bar_ref = useRef(null)
+    const [mounted, setMounted] = useState(false)
+    const [has_scrolled, setHasScrolled] = useState(false)
+    const [active_dropdown, setActiveDropdown] = useState('')
+    const [active_link_ref, setActiveLinkRef] = useState(null)
+    const [active_dropdown_ref, setActiveDropdownRef] = useState(null)
+    const [show_button, showButton, hideButton] = moveButton()
+    const current_page = useActiveLinkState('main')
+
+    const signup_url = is_ppc_redirect ? '/landing/signup/' : '/signup/'
+
+    const buttonHandleScroll = useCallback(() => {
+        setHasScrolled(true)
+        handleScroll(showButton, hideButton)
+    }, [])
+
+    const checkActive = (link_name) => link_name === active_dropdown || link_name === current_page
+
+    const handleLinkClick = (dropdown, target) => {
+        setActiveDropdown(dropdown)
+        if (!target) return
+        setActiveLinkRef(target)
+    }
+
+    const setDropdownRef = (new_ref) => setActiveDropdownRef(new_ref)
+
+    useOutsideClick(navigation_bar_ref, () => setActiveDropdown(''), active_dropdown_ref)
+
+    useEffect(() => {
+        setMounted(true)
+        document.addEventListener('scroll', buttonHandleScroll, { passive: true })
+        return () => document.removeEventListener('scroll', buttonHandleScroll)
+    }, [])
```

Comment:
this will throws exhaustive deps warning when checking code with eslint
because you are using setMounted inside useEffect but didn't add it to deps array